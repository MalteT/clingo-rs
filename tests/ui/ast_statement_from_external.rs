use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term = ast::Term::from(sym);
    let ext = ast::External::new(term, &[]);
    let stmt = ext.ast_statement();
    drop(ext);
    let _end = stmt;
}