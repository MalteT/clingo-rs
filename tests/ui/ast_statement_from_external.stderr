error[E0506]: cannot assign to `ext` because it is borrowed
  --> $DIR/ast_statement_from_external.rs:9:5
   |
7  |     let stmt = ext.ast_statement();
   |                --- borrow of `ext` occurs here
8  |     
9  |     ext = ast::External::new(term, &[]);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `ext` occurs here
10 |     let _end = (stmt, ext);
   |                 ---- borrow later used here
